### PR TITLE
Update to work with OpenTelemetry-Java 0.5.0

### PR DESCRIPTION
**Pull request by jkwatson**
changes:
* SpanData is now an interface, with a TestSpanData implementation for testing
* Update example code for Metric Instrument refinements
* Rename of the BatchSpanProcessor, and re-introduction of the builder

Note: we'll want to update the auto-instrumentation support when that gets released, probably next week. This will include adding support for auto-configuration of metric export.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ Important: If you are using [auto-instrumentation](#auto-instrumentation), you s
 
 2. Build the OpenTelemetry `BatchSpansProcessor` with the `NewRelicSpanExporter` 
 ```java
-    BatchSpansProcessor spanProcessor = BatchSpansProcessor.create(exporter);
+    BatchSpanProcessor spanProcessor = BatchSpanProcessor.newBuilder(exporter).build();
```
Comment (breedx-nr) [inline README.md:28]: I think `builder()` is more idiomatic than `newBuilder()` but don't feel too strongly about it.

Comment (breedx-nr) [reply README.md:28]: Hah, commented this before realizing that this is from upstream.  Derp.  I should have made this comment over there weeks ago it seems.  Heh.

Comment (jkwatson) [reply README.md:28]: months, actually. :)

Comment (jkwatson) [reply README.md:28]: For reasons I don't know, OTel has standardized on "newBuilder"

**Pull request merged**